### PR TITLE
JSON marshaling/unmarshaling of label Selector

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #16376 #15945

### CODE REVIEW

**Comment by wojtek-t** [comment]
@lavalamp - PTAL

File: pkg/fields/selector.go [hunk 1]
```diff
@@ -235,3 +236,26 @@ func parseSelector(selector string, fn TransformFunc) (Selector, error) {
 func OneTermEqualSelector(k, v string) Selector {
 	return &hasTerm{field: k, value: v}
 }
+
+// SelectorHolder is a wrapper around Selector that allows for
```
Comment (lavalamp) [inline pkg/fields/selector.go]: I was sure I'd already left a comment here, but I don't see it now. Maybe it's sitting unsent in one of my many open tabs.

Anyway, what do you think about moving these holders to pkg/api/unversioned, where we have other similar types?

Comment (wojtek-t) [inline pkg/fields/selector.go]: Done.

**Comment by lavalamp** [comment]
@k8s-oncall can you file some issues for the e2e tests that keep flaking in this PR?

**Comment by lavalamp** [comment]
LGTM

**Comment by lavalamp** [comment]
If you make a follow-up, I recommend changing the name to drop the "Holder" suffix on the type names-- it's not needed any more.

**Comment by lavalamp** [comment]
@k8s-bot test this

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please

**Comment by wojtek-t** [comment]
I've just done the rename requested above.

**Comment by lavalamp** [comment]
Thanks! still LGTM

**Pull request merged**